pkg/agent/core/ngt/handler/grpc/handler_test.go

```diff
@@ -4302,6 +4319,702 @@ func Test_server_MultiInsert(t *testing.T) {
+			args: args{
+				ctx:  ctx,
+				reqs: genSameReq(100, []float32{math.SmallestNonzeroFloat32, math.SmallestNonzeroFloat32, math.SmallestNonzeroFloat32}),
+			},
+			fields: fields{
+				name:              name,
+				ip:                ip,
+				svcCfg:            defaultF32SvcCfg,
+				svcOpts:           defaultSvcOpts,
+				streamConcurrency: 0,
+			},
+			want: want{
+				wantRes: genObjectLocations(100, name, ip),
+			},
+		},
+		{
+			name: "Boundary Value Testing case 3.1: Success to MultiInsert with max value vector (vector type is uint8)",
+			args: args{
+				ctx:  ctx,
+				reqs: genSameReq(100, []float32{math.MaxUint8, math.MaxUint8, math.MaxUint8}),
+			},
+			fields: fields{
+				name:              name,
+				ip:                ip,
+				svcCfg:            defaultIntSvcCfg,
+				svcOpts:           defaultSvcOpts,
+				streamConcurrency: 0,
+			},
+			want: want{
+				wantRes: genObjectLocations(100, name, ip),
+			},
+		},
+		{
+			name: "Boundary Value Testing case 3.2: Success to MultiInsert with max value vector (vector type is float32)",
+			args: args{
+				ctx:  ctx,
+				reqs: genSameReq(100, []float32{math.MaxFloat32, math.MaxFloat32, math.MaxFloat32}),
+			},
+			fields: fields{
+				name:              name,
+				ip:                ip,
+				svcCfg:            defaultF32SvcCfg,
+				svcOpts:           defaultSvcOpts,
+				streamConcurrency: 0,
+			},
+			want: want{
+				wantRes: genObjectLocations(100, name, ip),
+			},
+		},
+		func() test {
+			req := genIntReq(vector.Gaussian, 100, intVecDim)
+			req.Requests[0].Vector.Id = ""
+
+			uuids := make([]string, 0, len(req.Requests))
+			for _, r := range req.Requests {
+				uuids = append(uuids, r.Vector.Id)
+			}
+
+			return test{
+				name: "Boundary Value Testing case 4.1: Fail to MultiInsert with 1 request with empty UUID (vector type is uint8)",
+				args: args{
+					ctx:  ctx,
+					reqs: req,
+				},
+				fields: fields{
+					name:              name,
+					ip:                ip,
+					svcCfg:            defaultIntSvcCfg,
+					svcOpts:           defaultSvcOpts,
+					streamConcurrency: 0,
+				},
+				want: want{
+					err: status.WrapWithInvalidArgument(fmt.Sprintf("MultiInsert API invalid uuids \"%v\" detected", uuids), errors.ErrUUIDNotFound(0),
+						&errdetails.RequestInfo{
+							RequestId:   strings.Join(uuids, ", "),
+							ServingData: errdetails.Serialize(req),
+						},
+						&errdetails.BadRequest{
+							FieldViolations: []*errdetails.BadRequestFieldViolation{
+								{
+									Field:       "uuid",
+									Description: errors.ErrUUIDNotFound(0).Error(),
+								},
+							},
+						},
+						&errdetails.ResourceInfo{
+							ResourceType: ngtResourceType + "/ngt.MultiInsert",
+							ResourceName: fmt.Sprintf("%s: %s(%s)", apiName, name, ip),
+						}),
+				},
+			}
+		}(),
+		func() test {
+			req := genF32Req(vector.Gaussian, 100, f32VecDim)
+			req.Requests[0].Vector.Id = ""
```

Comment:
How about refactor `genF32Req` for set no uuid with-param.
e.g. `genF32Req(vec, num, dim, noid)`
when sid is true, set uuid as "".

Reply:
Becasue `genF32Req` will generate multiple insert requests, it is hard to decide which one needs empty ID and which one needs ID generation, I prefer the current style.

Reply:
the kinds of `create request` function increases, so I thinks it would be better rename about create function.
e.g.
`genF32Req` -> `genF32InsertReq`
or
combine `genF32Req` and `genIntReq` to the `genInsertReq` with type option.

Reply:
The reason of separating genF32Req and genIntReq is because of the logic of generation is a little bit different (int request needs to cast to float32 to fit in our api). so I will rename it to genXXXInsertReq for now.

Reply:
I could not understand.
You implemented a new multiInsertRequest generator function.
It seems similar.

> The reason of separating genF32Req and genIntReq is because of the logic of generation is a little bit different (int request needs to cast to float32 to fit in our api). so I will rename it to genXXXInsertReq for now.

I think we only have to implement using `swicth-case`. Isn't it correct?

Reply:
sorry I misread the code. yes you are right.I will refactor it now.

Reply:
The difference between `genF32Req` and `genIntReq` is only a vector generator.